spanner/changestreams/src/main/java/com/example/spanner/changestreams/QueryChangeStreamSample.java

```diff
@@ -0,0 +1,175 @@
+      System.out.println("Executing change stream initial query.");
+      List<ChangeStreamRecord> initialQueryRecords = executeChangeStreamQueryAndPrint(
+          dbClient, changeStreamName, startTimestampStr, endTimestampStr, null);
+
+      System.out.println("Executing change stream partition queries.");
+      for (ChangeStreamRecord record : initialQueryRecords) {
+        // Executes a partition query to print data records that we just inserted.
+        if (record instanceof ChildPartitionsRecord) {
+          for (ChildPartition childPartition :
+              ((ChildPartitionsRecord) record).getChildPartitions()) {
+            executeChangeStreamQueryAndPrint(
+                dbClient, changeStreamName, startTimestampStr, endTimestampStr,
+                childPartition.getToken());
+          }
+        }
+      }
+    } catch (Exception e) {
+      System.out.println(e);
+    } finally {
+      spanner.close();
+    }
+  }
+
+  // Insert two rows into Singers table.
+  static void insertRows(DatabaseClient client, String tableName) {
+    client.write(
+        ImmutableList.of(
+        Mutation.newInsertOrUpdateBuilder(tableName)
+          .set("SingerId").to(1)
+          .set("FirstName").to("Melissa")
+          .set("LastName").to("Garcia")
+          .build(),
+        Mutation.newInsertOrUpdateBuilder(tableName)
+          .set("SingerId").to(2)
+          .set("FirstName").to("Dylan")
+          .set("LastName").to("Shaw")
+          .build()
+      )
+    );
+  }
+
+  // Execute a change stream query, return and print out the result records.
+  static List<ChangeStreamRecord> executeChangeStreamQueryAndPrint(
+      DatabaseClient dbClient, String changeStreamName, String startTimestamp,
+      String endTimestamp, String partitionToken) {
+    System.out.println("Executing a change stream query with: "
+        + "start_timestamp => " + startTimestamp
+        + ", end_timestamp => " + endTimestamp
+        + ", partition_token => " + partitionToken
+        + ", heartbeat_milliseconds => 5000"
+        + ", read_options => null");
+
+    final String query =
+        String.format("SELECT * FROM READ_%s ("
+          + "start_timestamp => @startTimestamp,"
+          + "end_timestamp => @endTimestamp,"
+          + "partition_token => @partitionToken,"
+          + "read_options => null,"
+          + "heartbeat_milliseconds => @heartbeatMillis"
+          + ")", changeStreamName);
+
+    final ResultSet resultSet =
+        dbClient
+          .singleUse()
+          .executeQuery(
+            Statement.newBuilder(query)
+              .bind("startTimestamp").to(Timestamp.parseTimestamp(startTimestamp))
+              .bind("endTimestamp").to(Timestamp.parseTimestamp(endTimestamp))
+              .bind("partitionToken").to(partitionToken)
+              .bind("heartbeatMillis").to(5000)
+              .build());
+
+    ChangeStreamRecordMapper changeStreamRecordMapper = new ChangeStreamRecordMapper();
```

Comment:
I still think that we don't need to map the records. We could just return raw structs from here. This would eliminate all the classes we created and the mapper from the PR as well.

I think we discussed this before, sorry I don't remember if we reached a decision. Could you remind me?

Thanks!

Reply:
Sorry for leaving the PR for a long time! So I remembered the last time we agreed that the it's helpful to have some models which can teach customer how to decode these structs, and I recall that was the reason that we moved the PR from java client to here. 

I agreed that it's cumbersome to have these models, but imagine if I am a customer who is interested in trying change stream, I need to think a lot about how to process the results once receiving those raw structs, and try to decode them based data change record structure on the user guide, it would be easier to get people start if they can just access the models.

Reply:
Found the comments from original PR: https://github.com/googleapis/java-spanner/pull/1565#discussion_r762654054

Reply:
Ah ok, let's keep it then!

Reply:
Considering we have merged change stream connector code into Beam repo, I think I can just import Beam and use the model.